Docker.md

```diff
@@ -0,0 +1,264 @@
+
+# Now create a runtime image and copy all the build artifacts from build image
+FROM mcr.microsoft.com/dotnet/aspnet:5.0
+WORKDIR /app
+COPY --from=build-env /app/out ./
+
+# Our runtime image will call dotnet Project.Name.API.dll on startup
+ENTRYPOINT ["dotnet", "Project.Name.API.dll"]
+
+```
+
+This docker file will do a 2 stage image build, first it will create a temporary image with .NET SDK called build-env, it will copy the source inside of that image and build it.
+
+Second stage will build the runtime image, this image will only contain the .NET runtime and the built project files.
+
+To create a docker image in your project, add a `Dockerfile` to the root of your solution, then build and run the image :
+
+```bash
+> docker build -t <image-name>
+> docker run --rm -d --name <container-name> <image-name> \
+    -p 8000:80 \
+    -e ASPNETCORE_ENVIRONMENT=Development
+```
+
+You can add additional environment variables for the docker container with more `-e` switches.
+
+## Running ASP.NET core with HTTPS inside of docker
+
+More information on this can be found [here](https://docs.microsoft.com/en-us/aspnet/core/security/docker-compose-https?view=aspnetcore-5.0), but a quick overview
+
+### Create a dev certificate for localhost
+
+On Windows :
+
+```ps1
+dotnet dev-certs https -ep $env:USERPROFILE/.aspnet/https/<certificate-name>.pfx -p crypticpassword
+dotnet dev-certs https --trust
+```
+
+On MacOS/Linux :
+
+```bash
+> dotnet dev-certs https -ep ${HOME}/.aspnet/https/<certificate-name>.pfx -p crypticpassword
+> dotnet dev-certs https --trust
+```
+
+### Pass HTTPS configuration to docker image
+
+On Windows :
+
+```ps1
+> $env:HOME=$env:USERPROFILE
+> docker run --rm -d --name <container-name> -p 8000:80 -p 8001:443 \
+        -e ASPNETCORE_ENVIRONMENT=Development \
+        -e ASPNETCORE_URLS="https://+;http://+" \
+        -e ASPNETCORE_HTTPS_PORT=8001 \
+        -e ASPNETCORE_Kestrel__Certificates__Default__Password="crypticpassword" \
+        -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/<certificate-name>.pfx \
+        -v ${HOME}/.aspnet/https:/https/ \
+        <image-name>
+```
+
+On MacOs/Linux
+
+```bash
+> docker run --rm -d --name <container-name> -p 8000:80 -p 8001:443 \
+        -e ASPNETCORE_ENVIRONMENT=Development \
+        -e ASPNETCORE_URLS="https://+;http://+" \
+        -e ASPNETCORE_HTTPS_PORT=8001 \
+        -e ASPNETCORE_Kestrel__Certificates__Default__Password="<crypticpassword>" \
+        -e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/<certificate-name>.pfx \
+        -v ${HOME}/.aspnet/https:/https/ \
+        <image-name>
+```
+
+You should now be able to access HTTPS version of your site at `https://localhost:8001`.
+
+## Publishing docker images
+
+Docker images can be published to a [registry](https://docs.docker.com/registry/) - this is how you deploy an image - you publish it to a registry and then you reference the registry image on the server you want to start the image.
```

Comment:
FROM:
Docker images can be published to a [registry](https://docs.docker.com/registry/) - this is how you deploy an image - you publish it to a registry and then you reference the registry image on the server you want to start the image.
TO:
Docker images can be published to a [registry](https://docs.docker.com/registry/). To deploy an image you publish it to a registry and then you reference the registry image on the server you want to start the image.